### DISCUSSION TITLE
Help with online editor for other species

### DISCUSSION THREAD

**Discussion by annerilotter** [gosling-lang/gosling.js#919] [Q&A] [answered]
Hi

I love this project and the visualizations that can be created with it.

I wanted to ask if you could provide some guidance to generate a 'plain' circos for a non-reference organism using the online editor (I understand that you may not support this at the moment, but I would love if you could). I am struggling to understand how to set up the input data to generate a circus with just the chromosomes represented. What I have is as following: 

Chromosome,chromStart,chromEnd,Name
Chr01,0,44965197,Chr01
Chr02,0,59529170,Chr02
Chr03,0,83952244,Chr03
Chr04,0,41160059,Chr04
Chr05,0,76243064,Chr05
Chr06,0,57472304,Chr06
Chr07,0,54830020,Chr07
Chr08,0,72515979,Chr08
Chr09,0,39307835,Chr09
Chr10,0,37777128,Chr10
Chr11,0,44836791,Chr11

When I use the URL for this data, the image does not render the new data as input. Am I maybe missing something?

Kind regards

**Comment by sehilyi** [comment, marked as answer]
Hi @annerilotter! Thank you for the kind word and for visiting here to ask a question. And sorry for the a bit delayed response!

The use of other species is supported by specifying the custom chromosome names and lengths using `assembly`. I created a spec that works with your example that you can refer to.

## Spec

```ts
{
  "assembly": [
    ["Chr01", 44965197],
    ["Chr02", 59529170],
    ["Chr03", 83952244],
    ["Chr04", 41160059],
    ["Chr05", 76243064],
    ["Chr06", 57472304],
    ["Chr07", 54830020],
    ["Chr08", 72515979],
    ["Chr09", 39307835],
    ["Chr10", 37777128],
    ["Chr11", 44836791]
  ],
  "layout": "circular",
  "centerRadius": 0.6,
  "views": [
    {
      "spacing": 1,
      "tracks": [
        {
          "data": {
            "type": "csv",
            "url": "https://gist.githubusercontent.com/sehilyi/feaf4c87017ead51978353857d91711c/raw/0d8abf56cb62f2a1d56a07a49c34864f0d8ecfb6/annerilotter.csv",
            "chromosomeField": "Chromosome",
            "genomicFields": [
              "chromStart",
              "chromEnd"
            ]
          },
          "mark": "rect",
          "x": { "field": "chromStart", "type": "genomic"},
          "xe": { "field": "chromEnd", "type": "genomic"},
          "color": { "field": "Name", "type": "nominal"}
        }
      ]
    }
  ]
}
```

## Screenshot
![gosling-visualization (14)](https://github.com/gosling-lang/gosling.js/assets/9922882/9a182990-e1c1-4abb-807b-5851a0f3c681)

## Link to Editor
<a href="https://gosling.js.org/?full=false&spec=(B'assemblyPBQ0z965197T2A59529170T3A83952244T4A41160059T5A76243064T6A57472304T7A54830020T8A72515979T9A39307835-10A37777128-1z836791Vq'layoutjcircular'IO'centerRadiusD0.6I'viewsPB*(YOL'tracksPY*(YL'dataD(YL*JZ'urljhttps%3A%2F%2Fgist.githubusercontent.cU%2Fsehilyi%2Ffeaf4c87017ead51978353857d91711c%2Fraw%2F0d8abf56cb62f2a1d56a07a49c34864f0d8ecfb6%2Fannerilotter.ZHosUeFWjChrUosUe'G*'MFWsPX~'VLL)G'markjrect'G'xKX_xeK~A_colorK'NameAJnUinal')Y*)YV*)B%5D%0A)*%20%20-qQA'%2C%20B%0A*D!%20GILLH'chrUI%2CBJ'typejKD('fWDL**MgenUicO'spacingD1IPD%5BQ*%5B'ChrT-0UomV%5DBWieldXHStartAYBLZcsv'G*_JM')G'jD'q%5DIz1A44~HEnd%01~zqj_ZYXWVUTQPOMLKJIHGDBA-*_">Link</a>

**Comment by annerilotter** [comment]
Hi

Thank you so much for this, it really has made my day. I was really
struggling to understand this from the documentation and guidelines (maybe
I missed where this is specified or an example use case).

Kind regards

On Wed, 21 Jun 2023, 18:39 sehilyi, ***@***.***> wrote:

> Hi @annerilotter <https://github.com/annerilotter>! Thank you for the
> kind word and for visiting here to ask a question. And sorry for the a bit
> delayed response!
>
> The use of other species is supported by specifying the custom chromosome
> names and lengths using assembly. I created a spec that works with your
> example that you can refer to.
> Spec
>
> {
>   "assembly": [
>     ["Chr01", 44965197],
>     ["Chr02", 59529170],
>     ["Chr03", 83952244],
>     ["Chr04", 41160059],
>     ["Chr05", 76243064],
>     ["Chr06", 57472304],
>     ["Chr07", 54830020],
>     ["Chr08", 72515979],
>     ["Chr09", 39307835],
>     ["Chr10", 37777128],
>     ["Chr11", 44836791]
>   ],
>   "layout": "circular",
>   "centerRadius": 0.6,
>   "views": [
>     {
>       "spacing": 1,
>       "tracks": [
>         {
>           "data": {
>             "type": "csv",
>             "url": "https://gist.githubusercontent.com/sehilyi/feaf4c87017ead51978353857d91711c/raw/0d8abf56cb62f2a1d56a07a49c34864f0d8ecfb6/annerilotter.csv",
>             "chromosomeField": "Chromosome",
>             "genomicFields": [
>               "chromStart",
>               "chromEnd"
>             ]
>           },
>           "mark": "rect",
>           "x": { "field": "chromStart", "type": "genomic"},
>           "xe": { "field": "chromEnd", "type": "genomic"},
>           "color": { "field": "Name", "type": "nominal"}
>         }
>       ]
>     }
>   ]}
>
> Screenshot
>
> [image: gosling-visualization (14)]
> <https://user-images.githubusercontent.com/9922882/247627137-9a182990-e1c1-4abb-807b-5851a0f3c681.png>
> Link to Editor
>
> Link
> <https://gosling.js.org/?full=false&spec=(B'assemblyPBQ0z965197T2A59529170T3A83952244T4A41160059T5A76243064T6A57472304T7A54830020T8A72515979T9A39307835-10A37777128-1z836791Vq'layoutjcircular'IO'centerRadiusD0.6I'viewsPB*(YOL'tracksPY*(YL'dataD(YL*JZ'urljhttps%3A%2F%2Fgist.githubusercontent.cU%2Fsehilyi%2Ffeaf4c87017ead51978353857d91711c%2Fraw%2F0d8abf56cb62f2a1d56a07a49c34864f0d8ecfb6%2Fannerilotter.ZHosUeFWjChrUosUe'G*'MFWsPX~'VLL)G'markjrect'G'xKX_xeK~A_colorK'NameAJnUinal')Y*)YV*)B%5D%0A)*%20%20-qQA'%2C%20B%0A*D!%20GILLH'chrUI%2CBJ'typejKD('fWDL**MgenUicO'spacingD1IPD%5BQ*%5B'ChrT-0UomV%5DBWieldXHStartAYBLZcsv'G*_JM')G'jD'q%5DIz1A44~HEnd%01~zqj_ZYXWVUTQPOMLKJIHGDBA-*_>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gosling-lang/gosling.js/discussions/919#discussioncomment-6244017>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVLZ2FMB3N5D7BZTGXRNGULXMMPTFANCNFSM6AAAAAAZLVS45I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
This message and attachments are subject to a disclaimer.

Please refer to 
http://upnet.up.ac.za/services/it/documentation/docs/004167.pdf 
<http://upnet.up.ac.za/services/it/documentation/docs/004167.pdf> for
full 
details.
